### PR TITLE
Update codeowners post adapters-decoupling

### PR DESCRIPTION
**Pull request by MichelleArk**
resolves #N/A

### Problem

CODEOWNERS was referencing paths / teams that don't exist under dbt-labs!

### Solution

Update CODEOWNERS to reflect codebase.
### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [ ] I have run this code in development and it appears to resolve the stated issue  
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX
- [ ] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -15,28 +15,8 @@
 
 ### ADAPTERS
 
-# Adapter interface ("base" + "sql" adapter defaults, cache)
-/core/dbt/adapters              @dbt-labs/core-adapters
-
-# Global project (default macros + materializations), starter project
-/core/dbt/include               @dbt-labs/core-adapters
```
Comment (MichelleArk) [inline .github/CODEOWNERS:22]: this still exists but is just the starter_project for dbt init, shouldn't be an adapters concern any longer

https://github.com/dbt-labs/dbt-core/tree/main/core/dbt/include

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -15,28 +15,8 @@
...
-### TESTS
-
-# Overlapping ownership for vast majority of unit + functional tests
-
-# Perf regression testing framework
-# This excludes the test project files itself since those aren't specific
-# framework changes (excluded by not setting an owner next to it- no owner)
-/performance                    @nathaniel-may
-/performance/projects
+/tests/adapter                  @dbt-labs/adapters
```
Comment (colin-rogers-dbt) [inline .github/CODEOWNERS:19]: I don't think this is even us anymore, ideally y'all can eventually delete these

**Review by colin-rogers-dbt — APPROVED** [review]
one comment otherwise LGTM

File: .github/CODEOWNERS (hunk 2 above)
Comment (MichelleArk) [reply .github/CODEOWNERS]: Agree! Tracked in https://github.com/dbt-labs/dbt-core/issues/9513

**Pull request merged**